gh-91126: Docs and tests for slotted dataclasses with `__init_subclass__` (#123342)
python/cpython: Doc/library/dataclasses.rst

   - *slots*: If true (the default is ``False``), :attr:`~object.__slots__` attribute
     will be generated and new class will be returned instead of the original one.
     If :attr:`!__slots__` is already defined in the class, then :exc:`TypeError`
     is raised.

    .. warning::
        Calling no-arg :func:`super` in dataclasses using ``slots=True``
        will result in the following exception being raised:
        ``TypeError: super(type, obj): obj must be an instance or subtype of type``.
        The two-arg :func:`super` is a valid workaround.
        See :gh:`90562` for full details.

    .. warning::

>        Having a base class with :meth:`~object.__init_subclass__` with parameters

Reviewer: ```suggestion
       Passing parameters to a base class :meth:`~object.__init_subclass__`
```
